pallets/parachain-staking/src/set.rs

```diff
@@ -76,10 +105,37 @@ impl<T: Ord> OrderedSet<T> {
 		self.0.binary_search(&value).is_ok()
 	}
 
+	/// Return if the set contains `value`
+	pub fn binary_search(&self, value: &T) -> Result<usize, usize> {
+		self.0.binary_search(&value)
+	}
+
+	/// Return if the set contains `value`
+	pub fn binary_search_by<'a, F>(&'a self, f: F) -> Result<usize, usize>
+	where
+		F: FnMut(&'a T) -> Ordering,
+	{
+		self.0.binary_search_by(f)
+	}
+
 	/// Clear the set
 	pub fn clear(&mut self) {
 		self.0.clear();
 	}
+
+	/// Clear the set
+	pub fn len(&self) -> usize {
+		self.0.len()
+	}
+
+	/// Clear the set
+	pub fn is_empty(&self) -> bool {
+		self.0.is_empty()
+	}
+
+	pub fn to_vec(self) -> Vec<T> {
```

Comment:
Should we just implement `From<Vec<T>>`?

Reply:
We actually have both.

* `impl<T: Ord> From<Vec<T>> for OrderedSet<T>` To get from a `Vec` to a `Set`
* `impl<T: Ord> From<OrderedSet<T>> for Vec<T>` to get from the `Set` to a `Vec`

When you try the to use into, the compiler sometimes is not able to derive `T` (e.g. where I used to_vec()). That's why I added `to_vec`. Since `T` is already set here, we don't need to specify it when we would call `into`.

Reply:
But that's what turbofish is for, right? Something like `obj.into::<Vec<T>>()`?

Reply:
Yep. But I'm too lazy to explicitly state the type that I want to convert into. 😁 

It's probably named wrong. Should be `into_vec`. You can compare that with the [BinaryHeap::into_vec](https://doc.rust-lang.org/std/collections/struct.BinaryHeap.html#method.into_vec). They even use `into` in `into_vec` 😂 . It's just a short hand, so that you don't have to specify the type again.